docs/adr/0007-backup-policy.md

```diff
@@ -0,0 +1,23 @@
+# 7. Retain database backup for 90 days
+
+Date: 2023-02-08
+
+## Status
+
+Accepted
+
+## Context
+
+The database for job-server is currently stored on a managed cluster which is able to backups for seven days. A recent regression in the codebase caused some database fields to be truncated, and was not spotted until about six days after deployment. In this case, we were able to repair the truncated data using an old backup, but if it had gone a little bit longer without being spotted we would have experienced minor data loss.
+
+Our current opinion is that although seven days is sufficiently long enough to spot recover in the event of catastrophic dataloss, we should keep backups for longer to help recover from more subtle dataloss/degradation.
+
+The data stored in the database is not considered sensitive, and because of our desire & policy for openness a significant proportion of the contents are (or will become) public information.
+
+## Decision
+
+In addition to the seven-day backups provided by our database cluster, we will use an external service to retain backups of the jobserver database for a longer period. The backup will be generated daily, and we will keep the last 90 days of backups.
```

Comment:
I would be specific about the external service that we're using here.

Why the last 90 days of backups, out of interest? Why not 12 months?

Reply:
Yes, I'm not sure how to justify any specific length of time, as the thing we're trying to address is to give ourselves & end-users longer to notice data degradation due to accident. In the recent instance it took us 6 days, but it could just as easily take a year plus one day. 90 days is a significantly bigger window than we have now. The backup file is small, though, so we could make it longer? I don't really want to commit to keeping these backups in perpetuity.

Reply:
90 days seems reasonable. There's a risk to keeping data for longer, as we'd need to ensure we'd be able to delete user data if needed. It was Seb who originally mentioned 12 months, so if he's happy with 90 days that's fine.